google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryWatchTest.java

```diff
@@ -215,9 +213,9 @@ public void emptyResults_modifiedDocument_ADDED()
     ListenerAssertions listenerAssertions = listener.assertions();
     listenerAssertions.noError();
     listenerAssertions.eventCountIsAnyOf(Range.closed(2, 2));
-    listenerAssertions.addedIdsIsAnyOf(emptySet(), newHashSet("doc"));
-    listenerAssertions.modifiedIdsIsAnyOf(emptySet());
-    listenerAssertions.removedIdsIsAnyOf(emptySet());
+    listenerAssertions.addedIdsIsAnyOf(emptyList(), singletonList("doc"));
```

Comment:
This is surprising. Shouldn't it be one or the other deterministically?

Reply:
Yes. I fixed this instance, I think there are a couple other tests that could use tighter assertions. @BenWhitehead did go out of his way to make sure that these tests are not flaky, so I suspect some of these are intended.

Reply:
After thousands of iterations of these tests there is a fairly broad amount of wiggle room from the backend around event delivery and potential collection.  There are a number of internal SLOs that are monitored by the backend, but no SLA related to the events being delivered in a certain amount of time. Due to these observations, I added the any-of checks in several locations to reduce the flakiness of these tests. If there were publicly available SLAs for the event listener we could tighten up the tests bounds.

Reply:
Based on @BenWhitehead comment, I reverted this change.